spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -413,7 +435,11 @@ class OptimizeExecutor(
 
     val addFiles = txn.writeFiles(repartitionDF, None, isOptimize = true, Nil).collect {
       case a: AddFile =>
-        a.copy(dataChange = false)
+        (if (isClusteredTable) {
```

Comment:
I must miss something, where is the clustering domain metadata commited in this change?

Reply:
Domain metadata is committed when creating a clustered table [here](https://github.com/delta-io/delta/blob/b4e5d5c4532f9eb94edcbabe63e613619bf87003/spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala#L319).

This PR reads the clustering column from domain metadata using the [extractLogicalNames](https://github.com/delta-io/delta/pull/2461/files#diff-3773dfb6ca40a45d8407d111b056fb5c6acea3153eb59d05e3e0476bfc55a7a3R236) helper.